### PR TITLE
fix(ios): avoid crash when openStream fails

### PR DESCRIPTION
**Pull request by sgtcoolguy**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-27515

**Description:**
wraps in try/catch to properly set exception on JS context
if JS code is in async callback/event/timer and has no try/catch it will still crash
Also, differs from 'old' proxies in that the uncaught exception handler code isn't invoked

I'm not sure what the correct way to handle this is. This at least offers the user the ability to avoid the crash. If the code is invoked in the main body of a js file, it'll pop the error handler/controller with the stack trace.

But if it's invoked on an event listener it will:
- crash if not surrounded by try/catch
- let the user handle the error if caught by try/catch

I don't think it's specific to this PR, but we should investigate having the "unhandled" case on an event listener/callback/timer also show the error controller like the "old proxies" (the ones using the JS C API, not the Obj-C API) do.

I thought we had tried to interject an unhandled exception handler in the Obj-C API before and had to roll it back because it was affecting the mocha test suite's timer behavior (See #10862)

### CODE REVIEW

**Comment by sgtcoolguy** [comment]
I've pushed a couple follow-on commits to address some of the test failures once we get past the crash. Much like on Android, iOS uses a special class to handle encrypted files (TiFilesystemBlobProxy, equivalent to Android's TiResourceFile). This implementation didn't have some methods at all, also didn't have implementations of some we could feasibly do. I fixed those issues.

Haven't looked at the Blob failures yet, I assume something similar there.

File: iphone/Classes/FilesystemModule.m [hunk 1]
```diff
@@ -171,16 +180,21 @@ - (JSValue *)getFile
...
   NSString *newpath = [self pathFromComponents:args];
+  TiFile *fileProxy = [self getFileProxy:newpath];
+  return [self NativeToJSValue:fileProxy];
+}
 
-  if ([newpath hasPrefix:[self resourcesDirectory]] && ([newpath hasSuffix:@".html"] || [newpath hasSuffix:@".js"] || [newpath hasSuffix:@".css"] || [newpath hasSuffix:@".json"])) {
-    NSURL *url = [NSURL fileURLWithPath:newpath];
+- (TiFile *)getFileProxy:(NSString *)path
+{
+  if ([path hasPrefix:[self resourcesDirectory]] && ([path hasSuffix:@".html"] || [path hasSuffix:@".js"] || [path hasSuffix:@".css"] || [path hasSuffix:@".json"])) {
```
Comment (garymathews) [inline iphone/Classes/FilesystemModule.m]: I believe we can remove the suffix checks for `html`, `css` since we don't encrypt those.

**Review by garymathews — APPROVED** [review]
CR: **PASS**

**Comment by lokeshchdhry** [comment]
FR Passed.

`Ti.Filesystem.openStream` does not cause a crash with `deploytype` test.

Studio Ver:       5.1.4.201909061933
SDK Ver:          8.3.0 local build
OS Ver:           10.14.5
Xcode Ver:        Xcode 11.1
Appc NPM:         4.2.15
Appc CLI:         7.1.2-master.3
Daemon Ver:       1.1.3
Ti CLI Ver:       5.2.2
Alloy Ver:        1.14.1
Node Ver:         12.13.0
NPM Ver:          6.12.0
Java Ver:         10.0.2
Simulator:        Iphone 11 pro 13.1

**Pull request merged**